File: auth-center/auth-center-domain/src/main/java/com/jmsoftware/maf/authcenter/permission/service/PermissionDomainService.java
package com.jmsoftware.maf.authcenter.permission.service;

import com.baomidou.mybatisplus.extension.service.IService;
import com.jmsoftware.maf.authcenter.permission.persistence.Permission;
import com.jmsoftware.maf.common.domain.authcenter.permission.PermissionType;
import org.springframework.validation.annotation.Validated;

import javax.validation.constraints.NotEmpty;
import java.util.List;

/**
 * <h1>PermissionDomainService</h1>
 * <p>
 * Service of Permission.(Permission)
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com
 * @date 5 /11/20 8:34 AM
 */
@Validated
public interface PermissionDomainService extends IService<Permission> {
    /**
     * Gets permission list by role id list.
     *
     * @param roleIdList         the role id list
     * @param permissionTypeList the permission type list
     * @return the permission list by role id list
     */
    List<Permission> getPermissionListByRoleIdList(
            @NotEmpty List<Long> roleIdList,
            @NotEmpty List<PermissionType> permissionTypeList
    );
}

File: auth-center/auth-center-domain/src/main/java/com/jmsoftware/maf/authcenter/permission/service/PermissionDomainService.kt
package com.jmsoftware.maf.authcenter.permission.service

import com.baomidou.mybatisplus.extension.service.IService
import com.jmsoftware.maf.authcenter.permission.persistence.Permission
import com.jmsoftware.maf.common.domain.authcenter.permission.PermissionType
import org.springframework.validation.annotation.Validated
import javax.validation.constraints.NotEmpty

/**
 * # PermissionDomainService
 *
 * Domain Service of Permission. (Permission)
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/10/22 11:50 AM
 */
@Validated
interface PermissionDomainService : IService<Permission> {
    /**
     * Gets permission list by role id list.
     *
     * @param roleIdList         the role id list
     * @param permissionTypeList the permission type list
     * @return the permission list by role id list
     */
    fun getPermissionListByRoleIdList(
        roleIdList: @NotEmpty List<Long>,
        permissionTypeList: @NotEmpty List<PermissionType>
    ): List<Permission>
}
